### PR TITLE
WatchLaterPage

### PR DESCRIPTION
**Pull request by ShekharDhangar**
I have added WatchLater page. All p0 features are completed. Please review and let me if you find any bug or missing p0 feature.

### CODE REVIEW

File: src/context/playlist-context.js [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
+//     useContext,
+//     useState,
+//     useEffect,
+//   } from "react";
+// import { useLocalStorageGetItem } from "../hooks/customHooks";
+//   import { getPlaylists } from "../serverCalls/likeCalls";
+//   const PlaylistsContext = createContext();
+//   function PlaylistsProvider({ children }) {
+//     const [Playlists, setPlaylists] = useState([]);
+//     const userToken=useLocalStorageGetItem("shed-video-token");
```
Comment (Maianki) [inline src/context/playlist-context.js]: Please remove commented code if it is not being used.

File: src/pages/Error404Page/Error404Page.jsx [hunk 2]
```diff
@@ -5,10 +5,11 @@ export function Error404Page() {
     const navigate = useNavigate();
     const [counter, setCounter] = useState(10);
     useEffect(() => {
-        counter > 0 && setTimeout(() =>setCounter(count=>count-1), 1000);
+        const timer = counter > 0 && setTimeout(() =>setCounter(count=>count-1), 1000);
         if(counter===0){
             navigate("/")
         }
+        return (()=>clearTimeout(timer));
```
Comment (Maianki) [inline src/pages/Error404Page/Error404Page.jsx:12]: Appreciation -  good that you are using the cleanup function to clear timer

**Review by Maianki — APPROVED** [review]
Good job Shekhar nothing much to suggest. Great work on UI and your code also follows all good coding practices.

File: src/components/MobileSidebar/MobileSidebar.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    id:2,
+    navIcon: <MdOutlineExplore className="icon size-xs" />,
+    navTxt:"Explore",
+    navLink:"/explore",
+  },
+  {
+    id:3,
+    navIcon: <MdOutlineVideoLibrary className="icon size-xs" />,
+    navTxt:"Library",
+    navLink:"/library",
```
Comment (anik31) [inline src/components/MobileSidebar/MobileSidebar.jsx:21]: appreciation - liked that you're using a separate sidebar for mobile

File: src/components/VideoCard/VideoCard.jsx [hunk 4]
```diff
@@ -1,41 +1,100 @@
...
+
+  const deleteBtnHandler = () => {
+    if (location.pathname === "/history") {
+      removeFromHistoryCall(videoDetails?._id, userState?.id, setHistory);
+    } else if (location.pathname === "/liked-videos") {
+      removeLikedVideoCall(videoDetails?._id, userState?.id, setLikes);
+    } else if (location.pathname === "/watchLater") {
+      removeFromWatchLaterCall(videoDetails?._id, userState?.id, setWatchLater);
+    }else{
+      removeVideoFromPlaylistCall(playlistId,videoDetails?._id,userState?.id,dispatch);
```
Comment (anik31) [inline src/components/VideoCard/VideoCard.jsx:38]: appreciation - this is a very nice implementation for using the same card at different pages. I will be implementing the same.

File: src/context/playlist-context.js [hunk 5]
```diff
@@ -1,38 +1,38 @@
...
-      })();
-    }, []);
+//     useEffect(() => {
+//       (async () => {
+//         try {
+//           if (userToken) {
+//             const PlaylistsResponse = await getPlaylists(userToken);
+//             if (PlaylistsResponse) {
+//               setPlaylists(PlaylistsResponse.data.playlists);
+//             }
```
Comment (anik31) [inline src/context/playlist-context.js]: suggestion - avoid pushing commented code

File: src/pages/AuthenticationPage/AuthenticationComp/AuthSignUp/AuthSignUp.jsx [hunk 6]
```diff
@@ -28,8 +28,8 @@ function AuthSignUp() {
     password2: "",
   });
 
-  const passwordExpression =
-  /^(?=.*\d)(?=.*[a-z]).{5,10}$/;
+  const passwordExpression = 
+  /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/;
```
Comment (anik31) [inline src/pages/AuthenticationPage/AuthenticationComp/AuthSignUp/AuthSignUp.jsx:32]: appreciation - good that you're doing validation

File: src/pages/PlaylistVideoList/PlaylistVideoList.jsx [hunk 7]
```diff
@@ -21,16 +21,17 @@ export function PlayListVideoList() {
...
             {foundPlaylist.videos?.map((video) => (
               <VideoCard
                 key={video._id}
-                MenuBtn={PlaylistMenuBtn}
+                MenuBtn={true}
                 videoDetails={video}
               />
             ))}
-            {foundPlaylist.videos.length === 0 && (
+            {foundPlaylist?.videos?.length === 0 && (
```
Comment (anik31) [inline src/pages/PlaylistVideoList/PlaylistVideoList.jsx:34]: appreciation - good that you're using optional chaining

**Review by anik31 — COMMENTED** [review]
Code is good & UI is awesome. Great work 💯

File: src/context/playlist-context.js (hunk 5 above)
Comment (ShekharDhangar) [reply src/context/playlist-context.js]: thanks will remove it

File: src/context/playlist-context.js (hunk 1 above)
Comment (ShekharDhangar) [reply src/context/playlist-context.js]: ok

**Pull request merged**